uast/utils.go

```diff
@@ -18,7 +18,12 @@ func Tokens(n *Node) []string {
 	return tokens
 }
 
-// CyclomaticComplexity returns the cyclomatic complexity for the node. This uses the method of
+// CyclomaticComplexity returns the cyclomatic complexity for the node. The cyclomatic complexity
+// is a quantitative measure of the number of linearly independent paths through a program's source code.
+// It was developed by Thomas J. McCabe, Sr. in 1976. For a formal description see:
+// https://en.wikipedia.org/wiki/Cyclomatic_complexity
+// And the original paper: http://www.literateprogramming.com/mccabe.pdf
+// This implementation uses the method of
```

Comment:
- There is only one definition, McCabe, based on cyclic graphs which maybe we could implement if one day we implement an UAST->cyclic graph converter. This definition speaks about some things like shortcut boolean tests but not about all else.
- Then there are a lot of different implementations. I speak about this in the comment with three linked examples. I choose these tokens because if you check many implementations you'll see that is the most common one.
- I'll add an explanation about the counting of boolean operator in the current implementation.